Products/zms/rest_api.py

```diff
@@ -221,11 +221,13 @@ def __call__(self, REQUEST=None, **kw):
             REQUEST.RESPONSE.setHeader('Content-Type',ct)
             REQUEST.RESPONSE.setHeader('Content-Disposition', 'inline;filename="%s.%s"'%((self.ids+['get'])[-1],ct.split('/')[-1]))
             if ct == 'application/json':
-                # Filtering data records that may contain not strings but objects.
+                # Filtering data records on JSON serialization:
+                # - only str, int, float are allowed as values
+                # - lists and dicts are not allowed as values (except for Blob fields)
                 try:
                     data = json.dumps(data)
                 except:
-                    data = [{k:i[k] for k in i.keys() if not k=='records'} for i in data]
+                    data = [{k:i[k] for k in i.keys() if isinstance(i[k],(str,int,float))} for i in data]
```

Comment:
you should leave the condition for k!=records, otherwise you may get into trouble with ZMSRecordSets